src/platform/options/program_option.cpp

```diff
@@ -44,7 +44,14 @@ void mo::ProgramOption::parse_arguments(
     int argc,
     char const* argv[])
 {
-    auto parsed_command_line = po::command_line_parser(argc, argv).options(desc).allow_unregistered().run();
+    // Passing argc and argv directly to command_line_parser() should do same thing. Unfortunately, when argc is 0, boost
+    // 1.67+ does not correctly construct the vector and throws a std::bad_alloc
+    std::vector<std::string> args;
+    for (int i = 1; i < argc; i++)
+    {
+        args.push_back(*(argv + i));
+    }
```

Comment:
You could avoid the loop:
```
if (argc) args.swap(std::vector{argv+1, argv+argc});
```

Reply:
or just `po::command_line_parser(std::max(argc, 1), argv)`